util/dvsim/Scheduler.py

```diff
@@ -98,17 +98,22 @@ def _dispatch(self, hms, old_results):
 
         while len(to_dispatch) < num_slots and self._queued:
             next_item = self._queued.pop(0)
-
             # Does next_item have any dependencies? Since we dispatch jobs by
             # target, we can assume that each of those dependencies appears
             # in old_results.
-            has_failed_dep = False
+            has_failed_dep = False if next_item.needs_all_dependencies_passing else True
```

Comment:
This seems kind of weird to me. The code means that if `next_item` has no dependencies, it cannot run if `needs_all_dependencies_passing` is true. Is that really what you meant, and what does that do?

Reply:
That is not the intention. Thanks rupert for pointing that out! Let me get another PR to fix this!

Reply:
Cool. I think you can just leave this as `has_failed_dep = False` and then remove the "else" branch below.

Reply:
I thought we still want to keep the `else` branch because - if we collect coverage and no test passed, we would prefer to kill the coverage collection. Does PR #4998 looks okay to you?